### ISSUE TITLE
buildbot try (try_vc=git) fails with "Data must not be unicode"

### ISSUE THREAD

**Issue by yhager** [buildbot/buildbot#4746]
Running a command like `buildbot try -w me -b <builder>` results in a stack trace before even connecting to the master:
```
using 'ssh' connect method
job created
Unhandled Error
Traceback (most recent call last):
  File "/home/yuval/.virtualenv/buildbot/lib/python3.7/site-packages/twisted/internet/defer.py", line 654, in _runCallbacks
    current.result = callback(current.result, *args, **kw)
  File "/home/yuval/.virtualenv/buildbot/lib/python3.7/site-packages/buildbot/clients/tryclient.py", line 907, in <lambda>
    d.addCallback(lambda res: deliver())
  File "/home/yuval/.virtualenv/buildbot/lib/python3.7/site-packages/buildbot/clients/tryclient.py", line 669, in deliverJob
    reactor.spawnProcess(pp, argv[0], argv, os.environ)
  File "/home/yuval/.virtualenv/buildbot/lib/python3.7/site-packages/twisted/internet/posixbase.py", line 340, in spawnProcess
    processProtocol, uid, gid, childFDs)
--- <exception caught here> ---
  File "/home/yuval/.virtualenv/buildbot/lib/python3.7/site-packages/twisted/internet/process.py", line 770, in __init__
    self.proto.makeConnection(self)
  File "/home/yuval/.virtualenv/buildbot/lib/python3.7/site-packages/twisted/internet/protocol.py", line 514, in makeConnection
    self.connectionMade()
  File "/home/yuval/.virtualenv/buildbot/lib/python3.7/site-packages/buildbot/clients/tryclient.py", line 512, in connectionMade
    self.transport.write(self.job)
  File "/home/yuval/.virtualenv/buildbot/lib/python3.7/site-packages/twisted/internet/process.py", line 927, in write
    self.pipes[0].write(data)
  File "/home/yuval/.virtualenv/buildbot/lib/python3.7/site-packages/twisted/internet/process.py", line 203, in write
    abstract.FileDescriptor.write(self, data)
  File "/home/yuval/.virtualenv/buildbot/lib/python3.7/site-packages/twisted/internet/abstract.py", line 349, in write
    raise TypeError("Data must not be unicode")
builtins.TypeError: Data must not be unicode
```

The patch is very simple:
```
diff --git a/myproject/version.py b/myproject/version.py
index 68d7165..61ad8a0 100644
--- a/myproject/version.py
+++ b/myproject/version.py
@@ -1,4 +1,4 @@
-__version__ = '1.0.0.19'
+__version__ = '1.0.0.20'
 __describe__ = ''
 __revision__ = ''
 __build_type__ = 'RelWithDebInfo'
```

buildbot version 2.2.0, twisted version 19.2.0

**Comment by ishaijsonar** [comment]
In python 3.6 returning job.encode() instead of job in `buildbot.clients.tryclient.creatJobFile` removes error message for me, but it still doesn't trigger the requested build.

**Comment by p12tic** [comment]
Fixed in #4886.
